packages/smart-contracts/scripts-create2/contract-setup/execute-contract-method.ts

```diff
@@ -0,0 +1,82 @@
+import SafeApiKit from '@safe-global/api-kit';
+import { Contract, Overrides, Wallet } from 'ethers';
+import { safeAdminArtifact } from '../../src/lib/';
+import { CurrencyTypes } from 'types/dist';
+import Safe, { EthersAdapter } from '@safe-global/protocol-kit';
+import { ethers } from 'ethers';
+
+const txServiceUrls: Record<string, string> = {
+  mainnet: 'https://safe-transaction-mainnet.safe.global/',
+  goerli: 'https://safe-transaction-goerli.safe.global/',
+  sepolia: 'https://safe-transaction-sepolia.safe.global/',
+  matic: 'https://safe-transaction-polygon.safe.global/',
+  celo: 'https://safe-transaction-celo.safe.global/',
+  xsai: 'https://safe-transaction-gnosis-chain.safe.global/',
+  bsc: 'https://safe-transaction-bsc.safe.global/',
+  'arbitrum-one': 'https://safe-transaction-arbitrum.safe.global/',
+  avalanche: 'https://safe-transaction-avalanche.safe.global/',
+  optimism: 'https://safe-transaction-optimism.safe.global/',
+  zksyncera: 'https://safe-transaction-zksync.safe.global/',
+};
+
+export const executeContractMethod = async ({
+  network,
+  contract,
+  method,
+  props,
+  txOverrides,
+  signer,
+  safeMode,
+}: {
+  network: string;
+  contract: Contract;
+  method: string;
+  props: any[];
+  txOverrides: Overrides;
+  signer: Wallet;
+  safeMode?: boolean;
+}): Promise<void> => {
+  if (safeMode) {
+    const safeAddress = safeAdminArtifact.getAddress(network as CurrencyTypes.VMChainName);
+    const txServiceUrl = txServiceUrls[network];
+    if (!safeAddress || !txServiceUrl) {
+      throw new Error(`Can't administrate using Safe on ${network}`);
+    }
+
+    const ethAdapter = new EthersAdapter({
+      ethers,
+      signerOrProvider: signer,
+    });
+    const safeService = new SafeApiKit({ txServiceUrl, ethAdapter });
+    const safeSdk = await Safe.create({ ethAdapter, safeAddress });
+
+    const safeTransactionData = [
+      {
+        to: contract.address,
+        data: contract.interface.encodeFunctionData(method, props),
+        value: '0',
+      },
+    ];
+    console.log(safeAddress);
+    const nonce = await safeService.getNextNonce(safeAddress);
+    const safeTransaction = await safeSdk.createTransaction({
+      safeTransactionData,
+      options: { nonce },
+    });
+    const safeTxHash = await safeSdk.getTransactionHash(safeTransaction);
+    const senderSignature = await safeSdk.signTransactionHash(safeTxHash);
+
+    await safeService.proposeTransaction({
```

Comment:
If I'm not wrong, we can expect some error at this step, and we should maybe catch the error in order to append the `network` to its message.

Reply:
We already have this in the main calling methods.
Example:
```
console.warn(`An error occurred during the setup of ChainlinkConversionPath on ${network}`);
console.warn(err);
```